File: python/Car.py
#Saya Amelia Zalfa Julianti dengan NIM 2203999 mengerjakan Latihan 4 mata kuliah DPBO untuk keberkahanNya maka saya tidak melakukan kecurangan sesuai dengan apa yang telah dispesifikasikan. Aamiin.
from Vehicle import Vehicle

class Car (Vehicle):
    __jumlahKursi = ""
    __jumlahPintu= ""

    def __init__(self, plat = "", merk = "", tahunProduksi = "", warna = "", jumlahKursi = "", jumlahPintu = ""):
        
        super().__init__(plat, merk, tahunProduksi, warna)
        self.__jumlahKursi = jumlahKursi
        self.__jumlahPintu = jumlahPintu
         
    #Getter and Setter
    #Get plat
    def get_jumlahKursi(self):
        return self.__jumlahKursi
    
    #set plat
    def set_jumlahKursi(self, jumlahKursi):
        self.__jumlahKursi = jumlahKursi
    
    #Get merk
    
    def get_jumlahPintu(self):
        return self.__jumlahPintu
    
    #set jumlahPintu
    def set_jumlahPintu(self, jumlahPintu):
        self.__jumlahPintu = jumlahPintu
    

File: cpp/Car.cpp
#pragma once

//memasukkan llibrary yang digunakan, termasuk kelas induk (kelas Human) dari kelas Ssivitas  Akademink

#include <iostream>
#include <string>
#include "Vehicle.cpp"

//deklarasi namespace
using namespace std;

class Car : public Vehicle{
	private:


	string jumlahKursi;
	string jumlahPintu;

	public:

	//constructor

	Car(){

	}

	//Constructor : overloading
	Car(string jumlahKursi, string plat, string merk, string tahunProduksi, string warna) : Vehicle(plat, merk, tahunProduksi, warna){
		this->jumlahKursi = jumlahKursi;	
		jumlahPintu = "";
	}

	//constructor : overloading
	Car(string jumlahKursi, string jumlahPintu, string plat, string merk, string tahunProduksi, string warna) : Vehicle(plat, merk, tahunProduksi, warna){
		this->jumlahKursi = jumlahKursi;	
		this->jumlahPintu = jumlahPintu;
	}

	void setJumlahKursi(string jumlahKursi){
		this->jumlahKursi = jumlahKursi;
	}

	void setJumlahPintu(string jumlahPintu){
		this->jumlahPintu = jumlahPintu;
	}

	string getJumlahKursi(){
		return jumlahKursi;
	}

	string getJumlahPintu(){
		return jumlahPintu;
	}

	//destyrctors

	~Car(){

	}
};
